src/usbg.c

```diff
@@ -2011,6 +2011,70 @@ int usbg_create_function(usbg_gadget *g, usbg_function_type type,
 	return ret;
 }
 
+int usbg_get_interf_os_desc(usbg_function *f,
+			struct usbg_function_os_desc *f_os_desc)
+{
+	int ret = USBG_ERROR_NOT_SUPPORTED;
+	int nmb;
+	char spath[USBG_MAX_PATH_LENGTH];
+	const char *iname = f->ops->os_desc_iname;
```

Comment:
Hmmm.... Single function may have multiple interfaces (like CDC functions does). Also as far as I can see in kernel code function also reports number of interfaces as a param to usb_os_desc_prepare_interf_dir(). This means that we may have multiple interface.%s directories so I think that we should reflect this in our API

Reply:
The only function we support now (RNDIS) only uses a single interface. This is also hardcoded in the kernel, see de7a8d2d53 ("usb: gadget: f_rndis: OS descriptors support"). That is what I do in "libusbgx: Add interface name for Feature Descriptors".

I see what you are saying, we should add const char *iname to the usbg_get_interf_os_desc signature.

What I am wondering, since we add it to the public interface, should we make os_desc_iname also available in a public structure?

Reply:
I'm not sure but is it somehow useful for the user to know it?

Reply:
The standard OS descriptor interface name is a hardcoded/standardized string, e.g. in the RNDIS case:
http://elixir.free-electrons.com/linux/v4.14/source/drivers/usb/gadget/function/f_rndis.c#L942

If we add it to the usbg_(set/get)_interf_os_desc interface the user will have to provide it. Of course the user can hardcode it on its own too, but since we already have the value it would be useful for the user...

Reply:
Hm, about adding an getter:

        static char *usbg_get_os_desc_default_interf(usbg_function *f)
        {
                  return f->ops->os_desc_iname;
        }

Reply:
@kopasiak any thoughts? I guess we could add such a getter also later, and for now just update the signature of usbg_get/set_interf_os_desc.

Reply:
let's add it later